hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -217,9 +220,11 @@ def permitted(self, f):
         if self.width < 64:
             try:
                 float_of(f, self.width)
-                return True
             except OverflowError:
                 return False
+        smallest_normal = width_smallest_normals[self.width]
+        if (not self.allow_subnormal) and 0 < abs(f) < smallest_normal:
+            return False
         return True
```

Comment:
```suggestion
        return self.allow_subnormal or not (0 < abs(f) < width_smallest_normals[self.width])
```

Short-circuiting evaluation to get a performance win in the much-more-common case.

Reply:
Super awkward line for Black heh, hope you're happy with me removing the brackets lest we get
```python
        if (not self.allow_subnormal) and 0 < abs(f) < width_smallest_normals[
                self.width
        ]:
```